### ISSUE TITLE
Possible SQL Injection on QuerySession

### ISSUE THREAD

**Issue by RobSchoenaker** [JasperFx/marten#1170]
```
 private IReadOnlyList<TDoc> DoFullTextSearch<TDoc>(string searchTerm, string config, string searchFunction)
        {
            assertNotDisposed();
            var sql = $"where to_tsvector('{config}', data) @@ {searchFunction}('{config}', '{searchTerm}')"; // HERE
            var handler = new UserSuppliedQueryHandler<TDoc>(_store, sql, new object[0]);
            return _connection.Fetch(handler, _identityMap.ForQuery(), null, Tenant);
        }

        private Task<IReadOnlyList<TDoc>> DoFullTextSearchAsync<TDoc>(string searchTerm, string config, string searchFunction, CancellationToken token)
        {
            assertNotDisposed();
            var sql = $"where to_tsvector('{config}', data) @@ {searchFunction}('{config}', '{searchTerm}')"; // HERE
            var handler = new UserSuppliedQueryHandler<TDoc>(_store, sql, new object[0]);
            return _connection.FetchAsync(handler, _identityMap.ForQuery(), null, Tenant, token);
        }
```

Concatenating SQL this way is prone to SQL injection. Always use parameters i this case.

**Comment by oskardudycz** [comment]
@RobSchoenaker thank you for this suggestion. I'm right now doing improvements to FTS, I'll fix that while working on it (see more in [PR](https://github.com/JasperFx/marten/pull/1167) ). Good catch!

**Comment by RobSchoenaker** [comment]
I belive there are more places in the code like this. I will do a complete scan for all (non-theretical) possibilities and create new issues for them - if possible with a fix included.

**Comment by oskardudycz** [comment]
Great. Thank you in advance 👍

**Comment by mysticmind** [comment]
@oskardudycz has addressed the issue in #1167, Closing this.
